src/cudamatrix/cu-kernels.cu

```diff
@@ -2214,12 +2226,12 @@ static void _log_softmax_reduce(Real *y, const Real *x,
   __syncthreads();
 
   // Log sum step 1: substracts max, and takes exponentials.
-  y[threadIdx.x + j * d.stride] = x[threadIdx.x + j * d.stride] - max;
+  y[threadIdx.x + j * d.stride] = x[threadIdx.x + j * src_stride] - max;
   aux[threadIdx.x] = exp(y[threadIdx.x + j * d.stride]);
   for (int i = 1; i < steps; ++i) {
     if (threadIdx.x + i * THREADS < d.cols) {
       y[threadIdx.x + i * THREADS + j * d.stride] =
-        x[threadIdx.x + i * THREADS + j * d.stride] - max;
+        x[threadIdx.x + i * THREADS + j * src_stride] - max;
```

Comment:
It looks good now - `y` with `d.stride` and `x` with `src_stride`. I'm also working on this function, and will make a PR later.